packages/vue2/src/useForm.ts

```diff
@@ -36,19 +36,19 @@ export interface InertiaFormTrait {
   form<TForm>(rememberKey: string, data: TForm): InertiaForm<TForm>
 }
 
-export default function useForm<TForm>(data: TForm): InertiaForm<TForm>
-export default function useForm<TForm>(rememberKey: string, data: TForm): InertiaForm<TForm>
+export default function useForm<TForm>(data: TForm | (() => TForm)): InertiaForm<TForm>
+export default function useForm<TForm>(rememberKey: string, data: TForm | (() => TForm)): InertiaForm<TForm>
 export default function useForm<TForm>(...args): InertiaForm<TForm> {
   const rememberKey = typeof args[0] === 'string' ? args[0] : null
   const data = (typeof args[0] === 'string' ? args[1] : args[0]) || {}
   const restored = rememberKey ? (router.restore(rememberKey) as { data: any; errors: any }) : null
-  let defaults = cloneDeep(data)
+  let defaults = typeof data === 'object' ? cloneDeep(data) : cloneDeep(data())
```

Comment:
`useForm()` without any argument now throws an error, but it was a valid call before (in fact it's used widely in laravel)


It should be like this
```js
typeof data === 'function' ? cloneDeep(data()) :  cloneDeep(data)`
```